src/shared_realm.hpp

```diff
@@ -129,6 +134,18 @@ enum class SchemaMode : uint8_t {
     Manual
 };
 
+enum class ComputedPrivileges : uint8_t {
+    None = 0,
+    Read = (1 << 0),
+    Update = (1 << 1),
+    Delete = (1 << 2),
+    SetPermissions = (1 << 3),
+    Query = (1 << 4),
+    Create = (1 << 5),
+    ModifySchema = (1 << 6),
+    All = 0xFF
+};
```

Comment:
Would there be anything wrong with just using the `Privilege` enum from Sync?

Reply:
This abstracts away the difference between sync and non-sync builds so that bindings don't have to worry about that.